### PR TITLE
feat-0T170-20-create-migration-and-controller-base-of-testimonials

### PR DESCRIPTION
**Pull request by fabiianZ**
Crear modelo, migración y controlador base de Testimonials:

Campos:
name: VARCHAR NOT NULL
content: VARCHAR NULLABLE
timestamps y softDeletes
El modelo puede o no tener una imagen, y se debera utilizar Active Storage

Agregado estructuras bases de acuerdo a las especificaciones, . Instaladas gemas Discard y Active Storage

### CODE REVIEW

File: Gemfile
Comment (FernandoCiballero) [inline Gemfile:9]: Please add the gem below `dotenv`
```suggestion
gem 'dotenv-rails', '~> 2.7', '>= 2.7.6' # https://rubygems.org/gems/dotenv-rails
gem 'discard', '~> 1.2' # Please add the link as the other gems
gem 'jsonapi-serializer', '~> 2.2' # https://rubygems.org/gems/jsonapi-serializer
```

File: app/controllers/testimonials_controller.rb
Comment (FernandoCiballero) [inline app/controllers/testimonials_controller.rb:1]: Please check the description of the ticket, because it doesn't specify any endpoint, and you have to put  the controller into the namespace `api/v1`.

File: db/migrate/20220317235629_create_testimonials.rb
Comment (FernandoCiballero) [inline db/migrate/20220317235629_create_testimonials.rb:5]: Please check what datatypes are allowed by `active record`, and `null: true` is redundant in this case because is the default setting.

File: db/migrate/20220318120544_add_discarded_at_to_posts.rb
Comment (FernandoCiballero) [inline db/migrate/20220318120544_add_discarded_at_to_posts.rb:6]: Be careful not to `copy` and `paste` any code from the internet.

File: db/migrate/20220317235629_create_testimonials.rb
Comment (FernandoCiballero) [inline db/migrate/20220317235629_create_testimonials.rb:8]: Please check this line, because you are adding the `discarded_at` field in the other migration.

File: app/models/testimonial.rb
Comment (FernandoCiballero) [inline app/models/testimonial.rb:2]: Don't forget to add any validation. Please note that if you have a constraint into the database table, you need to validate the same into the model.

**Review by FernandoCiballero — CHANGES_REQUESTED** [review]
LGTM Fabian!!! 👏  I left some comments. Please avoid using `scaffold` in the project, instead of this use `rails g model` and/or `rails g controller`.

File: Gemfile
Comment (DevChristians) [inline Gemfile:12]: check alphabetic order on gem :p

File: app/controllers/api/v1/testimonials_controller.rb
Comment (DevChristians) [inline app/controllers/api/v1/testimonials_controller.rb:1]: I think you could delete this and keep the rest or delete the rest and keep this, because I think it's the same

File: app/models/testimonial.rb
Comment (DevChristians) [inline app/models/testimonial.rb:3]: add the validates rules
`validates :field, presence: true` 
for all fields
😄

File: config/routes.rb
Comment (DevChristians) [inline config/routes.rb:2]: apparently resources routes or anything route don't go yet

**Review by DevChristians — COMMENTED** [review]
I wait for the change for my approval, nice job!

File: Gemfile
Comment (FernandoCiballero) [reply Gemfile:11]: @fabiianZ please don't forget to change the format (with the link added in a commentary)

File: app/models/testimonial.rb
Comment (FernandoCiballero) [reply app/models/testimonial.rb:3]: I agree with @DevChristians, add the `presence: true` validation in the model field where you have a `null: false`.

File: db/migrate/20220318210131_create_testimonials.rb
Comment (FernandoCiballero) [inline db/migrate/20220318210131_create_testimonials.rb:4]: Just for the sake of respecting rails conventions, don't use uppercase words here.
```suggestion
      t.string :name, null: false
```

**Pull request merged**
